include/RAJA/pattern/forall.hpp

```diff
@@ -271,16 +294,20 @@ inline namespace policy_by_value_interface
  *
  ******************************************************************************
  */
-template <typename ExecutionPolicy, typename Res, typename IdxSet, typename LoopBody>
+template <typename ExecutionPolicy, typename Res, typename IdxSet, typename... Params>
 RAJA_INLINE resources::EventProxy<Res> forall_Icount(ExecutionPolicy&& p,
                                                      Res r,
                                                      IdxSet&& c,
-                                                     LoopBody&& loop_body)
+                                                     Params&&... params)
 {
   static_assert(type_traits::is_index_set<IdxSet>::value,
                 "Expected a TypedIndexSet but did not get one. Are you using "
                 "a TypedIndexSet policy by mistake?");
 
+  auto f_params = expt::make_forall_param_pack(std::forward<Params>(params)...);
+  auto loop_body = expt::get_lambda(std::forward<Params>(params)...);
```

Comment:
This should be by reference.
```suggestion
  auto&& loop_body = expt::get_lambda(std::forward<Params>(params)...);
```